Sources/CallWithDoxDialer/CallWithDoxDialer.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+#if canImport(UIKit)
+import UIKit
+
+/// A struct to handle dialing phone numbers using the Doximity app.
+public struct DoxDialerCaller {
+    /// Singleton instance of `DoxDialerCaller`.
+    public static let shared = DoxDialerCaller()
+
+    /// Constants used in `DoxDialerCaller`.
+    private enum Constants {
+        static let doximityScheme = "doximity://"
+        static let dialerTargetNumberPath = "dialer/call?target_number="
+        static let appsFlyerID = "id393642611"
+    }
+
+    private let application: OpenApplicationURL
+
+    init(application: OpenApplicationURL = UIApplication.shared) {
+        self.application = application
+    }
+
+    /// Dials a phone number using the Doximity app if it's installed; otherwise,
+    /// it directs the user to the App Store page for the Doximity app.
+    /// - Parameter phoneNumber: The 10-digit phone number `String` to dial
+    public func dialPhoneNumber(_ phoneNumber: String) {
+        if isDoximityInstalled {
+            guard let dialerURL = URL(string: "\(Constants.doximityScheme)\(Constants.dialerTargetNumberPath)\(phoneNumber)") else { return }
+            openURL(dialerURL)
+        } else {
+            guard let url = doximityAppStoreURL else { return }
+            openURL(url)
+        }
+    }
+
+    /// Returns the Doximity icon image.
+    public var doximityIcon: UIImage {
+        return iconFromPackage ?? iconFromBundle ?? UIImage()
```

Comment:
Should we throw an error her and let the consumers handle the case where we can't load the icon from the bundle?